keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepositoryTests.kt

```diff
@@ -177,70 +181,116 @@ abstract class ResourceRepositoryTests<T : ResourceRepository> : JUnit5Minutests
         }
       }
 
-      context("updating the state of the resource") {
-        context("an event that should be persisted in history") {
+      context("updating the event history of the resource") {
+        context("appending a resource event") {
           before {
             tick()
-            // TODO: ensure persisting a map with actual data
-            subject.appendHistory(ResourceUpdated(resource, emptyMap(), clock))
+            subject.appendHistory(ResourceUpdated(resource, mapOf("delta" to "some-difference"), clock))
           }
 
-          test("the event is included in the history") {
+          test("the event is included in the resource history") {
             expectThat(subject.eventHistory(resource.id))
               .hasSize(2)
               .first()
               .isA<ResourceUpdated>()
+              .get { delta }.isEqualTo(mapOf("delta" to "some-difference"))
+          }
+        }
+
+        context("appending an identical resource event with duplicates allowed") {
```

Comment:
I pulled these out from under the context because I find nested tests hard to reason about. I know these causes some duplication of code in setting up the tests in the `before` block, but I much prefer that than having to read through multiple nested contexts to understand what the setup for a particular test is.